### DISCUSSION TITLE
DS918+ working stably @5Gbps with DSM 7.21 and Wavlink RTL1857

### DISCUSSION THREAD

**Discussion by m4gr4th34** [bb-qq/r8152#440] [Show and tell]
Using this Wavlink adapter: 
https://www.amazon.com/WAVLINK-Ethernet-5000Mbps-Converter-Aluminum/dp/B0DBZ41BPL/ref=cm_cr_arp_d_product_top?ie=UTF8&th=1

And this UGreen USB-C-to-USB-A 10Gbps throughput: 
https://www.amazon.com/dp/B0CY1Y3TSQ?ref=ppx_yo2ov_dt_b_fed_asin_title

Using the front port, I followed the instructions exactly, including manually enabling driver on install and rebooting.  It also works with a similar 2.5Gbps adapter, but I'm sticking with the Wavlink for now.  

Thanks for making this, the speeds are nice!!

Stable over 24 hours.  Will update in a week or two.  Also, might do the same for a 218+ I have, though it might not need 5Gbps.

**Comment by Roendi61** [comment]
Witch Driver you take? Have the same Adapter Wavlink. 2.5 Gb works fine. But 5 GB will not working

**Reply by m4gr4th34** [reply]
I have tried both the pre-release version from last week, as well as this:

[r8152-apollolake-2.18.1-1_7.2.spk](https://github.com/bb-qq/r8152/releases/download/2.18.1-1/r8152-apollolake-2.18.1-1_7.2.spk)

The older driver I used worked stable for over a week now.  I cannot enable jumbo frames (large MTU).  The adapter gets very hot, which slows down transmission, so I got an SSD heatsink for it which helps, although it is still pretty warm.

I also just installed the latest driver (r8152-apollolake-2.18.1-2_7.2.spk), and it seems to be working at 5Gbps also.  I used the help article and confirmed that my NAS does support jumbo frames, the terminal responded "yes" to the input: /bin/get_key_value /etc.defaults/synoinfo.conf supportMTU.  But, even though I enabled SMB, set SMB min to "SMB2 + large MTU", and max SMB to SMB3; when I try to increase MTU (via network, edit), it still does not allow me to go past 1500.  So, the "pre-release" driver does not seem to be making any difference ... regarding MTU.

Strangely, using ethtool, synology says it is only advertising up to 2.5Gbps, but when I use crystaldisk on windows it does transfer up to 5Gbps. Also, when I look at the log on my router is shows the nas connected at 5Gbps (even though, under the status of the etherport is shows up as "0Gbps").   so, I'm not sure what is going on, but software in general seems to be buggy nowadays across all platforms (lazy, overpaid developers..?  hehe)

To troubleshoot:
1. did you plug the adapter in to the FRONT USB port of the NAS?  
2. what USB-A to USB-C adapter are you using (and does it support 10Gbps data rate)?  Wavlink only has USB-C connector, and DS918+ only has USB-A.

**Reply by TheS1R** [reply]
Try changing MTU from NAS command line: **sudo ifconfig eth2 mtu 9000**

**Reply by m4gr4th34** [reply]
I can confirm that using the CLI command above does indeed work in changing the MTU now, as confirmed using: ip a s eth2

However, increasing jumbo packets on windows to 9000 as well actually reduces performance a bit (reading from NAS drops from ~450MB/s to ~350+MB/s, using crystal disk (also confirmed transferring large video file back and forth (~20GB size).  Jumbo frames is enabled on router, and I have the TPLink 10GB switch that has it as well.

However, I will keep jumboframes enabled on the NAS for a few hours to see if the standby heat of the adapter is lower (maybe i'll save some electricity, it is like a little space heater right now).

**Reply by m4gr4th34** [reply]
Quick note, I use this heatsink with fan:  https://www.amazon.com/dp/B0088N7LE4?ref=ppx_yo2ov_dt_b_fed_asin_title&th=1
Along with this USB-A to Sata power adapter:  https://www.amazon.com/dp/B0D41ND6XY?ref=ppx_yo2ov_dt_b_fed_asin_title&th=1

It comes to an extra $20, but it runs MUCH cooler, and is completely silent, which should dramatically improve the lifespan of this adapter, since I really don't want to have i/o breakdown on my NAS...  Hope it's helpful.

**Comment by Roendi61** [comment]
Wavlink RTL1857 with 2.5 GB runs stable and without problems. I have now ordered the same USB to USC C converter. Everything else is exactly the same. I have also tested the front connector. The data throughput is slower than when I connect the Wavlink directly to a 2.5 GB port on the switch. This way I achieve around 270 Mbit. MTU 9000 also worked for me

**Reply by m4gr4th34** [reply]
ok, and you are 100% sure you are using the latest 7.2 version of DSM?  DSM does not auto-update to 7.2, I needed to perform the update manually (because synology could not guarantee risk-free upgrade from 7.1 to 7.2).

**Reply by m4gr4th34** [reply]
Also, what exactly do you mean by this sentence, I don't understand it:  "The data throughput is slower than when I connect the Wavlink directly to a 2.5 GB port on the switch."  I don't understand what this means?

**Reply by Roendi61** [reply]
Thanks for your feedback
I am currently using DSM 7.2.2-72806 on the NAS
If I use a 10 GB SFP adapter on my Switch, the connection speed is slower than if I use a 2.5 GB SFP adapter.
If I change the MTU size with:
sudo ifconfig eth2 mtu 9000
After a restart, the old MTU value is restored and finally, when I plug the Wavlink into my 10 GB SFP adapter, my network status is empty. I have attached a screenshot. With the 2.5 GB SFP adapter it shows the correct values. What I also notice. My 
Wavlink adapter is only hand warm.
Sorry for my bad English


![SFP 10 GB](https://github.com/user-attachments/assets/200ba0b3-eaf7-404b-affb-5673aa6933a8)

**Reply by Roendi61** [reply]
When I copy a large file, the data throughput with my 2.5 GB connection is approx. 280 MBit with the 10 GB SFP adapter, but only approx. 230 MBit and not over 300 like yours. But I'm already very happy with the current situation. My troubleshooting is going in the direction of the USB adapter. As reported by other users, the most obvious problem is

**Reply by m4gr4th34** [reply]
i think i understand a bit better.  i don't use any SFP in my setup, all ethernet.  I heard that SPF-ethernet adapters cause a lot of problems at 10G still, so maybe that is your problem...  But, let's see if the USB adapter fixes the problem.

**Comment by Roendi61** [comment]
Thanks for your support

I give soon as possible feedback.

Am So., 20. Okt. 2024 um 09:33 Uhr schrieb m4gr4th34 <
***@***.***>:

**Reply by Roendi61** [reply]
I have found a good High Speed usb to usb c Adapter. Nothing change. When I connect on the 2.5 GB Fiber Adapter it works fine. When I tried to connect to the 10 GB Fiber Adapter the Connection is very slow. But for me no Problem. With the 2.5 GB Speed I am very happy. The Network Status on the Connection will showing now correct. I can change the MTU over the GUI. All is working

**Comment by Roendi61** [comment]
I have gone through your error program in detail.
In the section 
Check your transceiver (if you are using SFP+)section, 
I noticed that my tablet can establish a 5GB connection with the same Waflink adapter. But the NAS has a TPLink SFP adapter in the switch and the tablet has a no-name adapter. So I connected the NAS to the NoName adapter and now the connection via the 10 GBit adapter also works. But the speed is still lower than when I use a 2.5 GB SFP adapter. 
From my point of view, it cannot be due to the USB A to C connection.
The lsusb tool shows this as desired:
|__2-2 0bda:8157:3000 00 3.20 5000MBit/s 544mA 1IF (Realtek USB 10/1
All this for your as information. I'll see if I can find an SFP adapter that also meets the specification for Linux.
Thanks for your hard work

I am very happy with the current Situation
